x/oracle/keeper/oracle.go

```diff
@@ -31,7 +31,11 @@ func (k Keeper) RegisterOracle(ctx sdk.Context, msg *types.MsgRegisterOracle) er
 
 	// TODO: Add active queue of Oracle Registration
 
-	// TODO: emit RegisterOracle event
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.EventTypeRegisterOracle,
+			sdk.NewAttribute(types.AttributeKeyRegisterOracle, types.AttributeValueRegisterOracle)),
```

Comment:
`sdk.NewAttribute` accepts key and value of the attribute that you want to emit. 

For example in [`send.go`](https://github.com/cosmos/cosmos-sdk/blob/71035879e4af4551b9bcf16f79bc52785793602e/x/bank/keeper/send.go#L209) of `x/bank` module, the recipient is one of the attributes whose key is `types.AttributeKeyRecipient` and the value is `toAddr.String()`.

In case of `MsgRegisterOracle`, I think the value of the attribute should be the address of oracle
```go
ctx.EventManager().EmitEvents(sdk.Events{
		sdk.NewEvent(
			types.EventTypeRegisterOracle,
			sdk.NewAttribute(types.AttributeKeyRegisterOracle, mgs.oracleAddress)),
})
```

Reply:
It seems that one of the information needed when detecting an event in subscribe is the `EventType`.
I think this EventType would be defined like this:
```
RegisterOracleEvent
RegisterOracleCompleteEvent
SellDataEvent
GetBuyDataEvent
RegisterOracleMigrationEvent
RegisterOracleMigrationCompleteEvent
```

Subscribers should see this event and decide what to do about it.
I think @inchori triggered an event to get this type. Is that right?

Reply:
If my prediction is correct, it would be better to fire an event like this:
```
vote.type=RegisterOracleEvent
vote.type=RegisterOracleCompleteEvent
vote.type=SellDataEvent
vote.type=GetBuyDataEvent
vote.type=RegisterOracleMigrationEvent
vote.type=RegisterOracleMigrationCompleteEvent
```
The name of `vote.type` deserves some thought.

Reply:
Yes, as we discussed, the `AttributeKey` of `Event` should be like that @gyuguen commented. If we defined the events like that, that can be detected easy for subscriber.